client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -254,38 +258,42 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder) {
 
     @Override
     public CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
+                                                  final Map<Double, Double> newKeyRanges, final long timeoutMillis,
```

Comment:
@fpj making `scalestream` longpoll will mean periodic checking on controller because the way it happens on controller is as follows:
1. receive manual scale
2. start scale by update stream state in metadata store
3. post event in event processor's requeststream for scale operation
4. event is processed asynchronously (potentially on another controller instance)
5. keep checking metadata records to determine if scale completes. 

if same controller was running the scale operation then we could have linked the response to completion of processing. Hence longpoll would have worked without requiring controller to spend addition compute cycles.
But since that is not the case, and we would need to keep doing (5) so i dont like the idea of long poll for manual scale operations. 
hence we shifted the logic of periodically checking for the status on controller. 
i like the idea of removing the timeout and returning an object that has cancellation semantics. 

Also, this is not the only operation which breaks the symmetry and has start and check status. 
A commit transaction is exactly similar. Calling commit merely puts an event into commitStream and returns. Committing happens in the background. And there is a checkTxn status api. 

I think we could use a similar 'blocking' api which can wait until commit completes in case applications want it.